master/buildbot/db/changes.py

```diff
@@ -250,6 +250,34 @@ def thd(conn):
     def _getDataFromRow(self, row):
         return row.changeid
 
+    def getChangesForBranch(self, branch, order, count):
+        def thd(conn):
+            changes_tbl = self.db.model.changes
+            bsss_tbl = self.db.model.buildset_sourcestamps
+
+            from_clause = changes_tbl.join(bsss_tbl,
+                                           changes_tbl.c.sourcestampid == bsss_tbl.c.sourcestampid)
+            if branch == 'all':
+                q = sa.select([changes_tbl.c.changeid.distinct()]).select_from(
+                    from_clause)
+            else:
+                q = sa.select([changes_tbl.c.changeid.distinct()]).select_from(
+                    from_clause).where(changes_tbl.c.branch == branch)
+            q = q.limit(count).order_by(changes_tbl.c.changeid.desc())
+            rp = conn.execute(q)
+            changeids = [self._getDataFromRow(row) for row in rp]
+            rp.close()
+            return list(changeids)
+
+        d = self.db.pool.do(thd)
```

Comment:
Use `yield` here and `@defer.inlineCallbacks` on getChangesForBranch.

Reply:
not sure if it's correct in the new form since I think it will only return the changeids. Or am I wrong? Any suggestions?